### DISCUSSION TITLE
/expo-router/ dependency issues

### DISCUSSION THREAD

**Discussion by bolongarof** [expo/expo#25564] [Archived: Questions]
I'm getting this error when trying to start my expo app. I have tried various versions of npm and of all the dependencies:

› Open in the web browser...
(node:5684) [DEP_WEBPACK_DEV_SERVER_CONSTRUCTOR] DeprecationWarning: Using 'compiler' as the first argument is deprecated. Please use 'options' as the first argument and 'compiler' as the second argument.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:5684) [DEP_WEBPACK_DEV_SERVER_LISTEN] DeprecationWarning: 'listen' is deprecated. Please use the async 'start' or 'startCallback' method.
› Press ? │ show all commands
› Open in the web browser...
› Press ? │ show all commands
WARNING in ./node_modules/expo-router/src/exports.ts:26
export 'ErrorBoundaryProps' (reexported as 'ErrorBoundaryProps') was not found in './views/Try' (possible exports: Try)
  24 | export { ExpoRoot } from "./ExpoRoot";
  25 | export { Unmatched } from "./views/Unmatched";
> 26 | export { ErrorBoundaryProps } from "./views/Try";
  27 | export { ErrorBoundary } from "./views/ErrorBoundary";
  28 |
  29 | // Platform

ERROR in ./node_modules/expo-router/_ctx.web.tsx:1:19
Module not found: Can't resolve '..\..\app'
> 1 | export const ctx = require.context(
    |                   ^
  2 |   process.env.EXPO_ROUTER_APP_ROOT!,
  3 |   true,
  4 |   /.*/,

web compiled with 1 error and 1 warning

My latest and greatest attempt package.json:

{
  "name": "logintest",
  "version": "1.0.0",
  "main": "expo-router/entry",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web"
  },
  "dependencies": {
    "@expo/webpack-config": "^19.0.0",
    "@react-navigation/bottom-tabs": "^6.5.11",
    "@react-navigation/drawer": "^6.6.6",
    "@types/react": "~18.2.14",
    "expo": "~49.0.13",
    "expo-constants": "~14.4.2",
    "expo-linking": "~5.0.2",
    "expo-router": "^2.0.0",
    "expo-status-bar": "~1.6.0",
    "react": "18.2.0",
    "react-native": "0.72.6",
    "react-native-gesture-handler": "~2.12.0",
    "react-native-popup-menu": "^0.16.1",
    "react-native-reanimated": "~3.3.0",
    "react-native-safe-area-context": "4.6.3",
    "react-native-screens": "~3.22.0",
    "react-native-svg": "13.9.0",
    "react-native-web": "~0.19.6",
    "typescript": "^5.1.3"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0",
    "@types/webpack-env": "^1.18.4",
    "react-native-svg-transformer": "^1.1.0",
    "sharp-cli": "^2.1.0",
    "source-map-support": "^0.5.21"
  },
  "private": true
}

Please help!

**Comment by SwapRane12** [comment]
Metro Support is not enabled in app, enable it as shown in [manual installation](https://docs.expo.dev/routing/installation/#manual-installation)
add this to your app.json. it will work
`{
  "web": {
    "bundler": "metro"
  }
}
`
